Sources/SourceControl/GitRepository.swift

```diff
@@ -16,13 +16,13 @@ import func POSIX.getenv
 import enum POSIX.Error
 
 enum GitRepositoryProviderError: Swift.Error {
-    case gitCloneFailure(url: String, path: AbsolutePath)
+    case gitCloneFailure(url: String, path: AbsolutePath, errorOutput: String)
```

Comment:
I was following along with GitRepositoryProviderError already being defined here as an extension, although I'm not entirely clear on what the conventions on for when you update here vs. `Commons/Error.swift`. Should I move the whole error enum extension and it's description over to `Common/Error.swift`? (i.e. lines 18 through 28 in this file?)